contracts/lib/LibPrices.sol

```diff
@@ -38,53 +38,91 @@ library Prices {
         pure
         returns (int256)
     {
-        return int256((averageTracerPrice - averageOraclePrice) / 90);
+        return
+            (averageTracerPrice.toInt256() - averageOraclePrice.toInt256()) /
+            90;
     }
 
+    /**
+     * @notice Calculate the average price of trades in a PriceInstant instance
+     * @param price Current cumulative price and number of trades in a time period
+     * @return Average price for given instance
+     */
     function averagePrice(PriceInstant memory price)
         public
         pure
         returns (uint256)
     {
+        // todo double check safety of this.
+        // average price == 0 is not neccesarily the
+        // same as no trades in average
         if (price.trades == 0) {
             return 0;
         }
         return price.cumulativePrice / price.trades;
     }
 
+    /**
+     * @notice Calculates average price over a time period of 24 hours
+     * @dev Ignores hours where the number of trades is zero
+     * @param prices Array of PriceInstant instances in the 24 hour period
+     * @return Average price in the time period (non-weighted)
+     */
     function averagePriceForPeriod(PriceInstant[24] memory prices)
         public
         pure
         returns (uint256)
     {
-        uint256 n = prices.length <= 24 ? prices.length : 24;
         uint256[] memory averagePrices = new uint256[](24);
 
-        for (uint256 i = 0; i < n; i++) {
+        // TODO: make sure this procedure is gas-optimised
```

Comment:
Is this something we still need to address?

Reply:
I don't think so. We could test if memory or storage for currPrice is cheaper but I think it's fine.

Reply:
I just wrote it since the `continue`/second-increment variable was the first thing that came to mind, and probably could be more effecient somehow